contracts/contracts/strategies/FraxConvexStrategy.sol

```diff
@@ -0,0 +1,477 @@
+    {
+        return CurveTwoCoinFunctions.getCurveFunctions();
+    }
+
+    /**
+     * @dev Unlock all the Frax Staked Convex LP tokens if:
+     * - there is a lock and
+     * - the lock has expired
+     * @return unlockedAmount The amount of Frax Staked Convex LP tokens unlocked.
+     * This will be zero if there is no lock or the lock has not yet expired.
+     */
+    function _unlock() internal returns (uint256 unlockedAmount) {
+        // If a lock exists and it has expired
+        if (lockKey != NO_KEY && block.timestamp > unlockTimestamp) {
+            // Withdraw all the Frax Staked Convex LP tokens from the lock
+            // to this strategy contract.
+            // Have to withdraw all as we can't withdraw a partial amount.
+            // The third `claim_rewards` parameter will still claim rewards even
+            // when set to `false`. It will only not collect rewards if the
+            // Frax Convex Locking contract has collectRewardsOnWithdrawalPaused = true.
+            unlockedAmount = IFraxConvexLocking(fraxLocking).withdrawLocked(
+                lockKey,
+                address(this),
+                false
+            );
+
+            emit Unlock(lockKey, unlockedAmount, unlockTimestamp);
+
+            // The previous withdraw deletes the old lock so reset
+            // slither-disable-next-line reentrancy-no-eth
+            lockKey = NO_KEY;
+        }
+        // else no lock exists or has not expired so nothing to do.
+    }
+
+    /**
+     * @dev Does not do anything if the locked balance is over the target locked balance.
+     * Else extends an existing lock for 7 days regardless of whether it has expired or not.
+     * Create a new lock if one doesn't exist and there is a positive target locked balance.
+     * Or add to the existing lock if the locked tokens is under the target locked balance.
+     * @param unlockedBalance The balance of the Frax Staked Convex LP tokens that are NOT locked.
+     * @param lockedBalance The balance of the Frax Staked Convex LP tokens that are locked.
+     * This includes any tokens in an expired lock.
+     */
+    function _lock(uint256 unlockedBalance, uint256 lockedBalance) internal {
+        // Read targetLockedBalance into memory to save a second SLOAD
+        uint256 targetLockedBalanceMem = targetLockedBalance;
+
+        // Do not extend lock time or add funds if already over the target locked balance
+        if (lockedBalance > targetLockedBalanceMem) {
+            return;
+        }
+
+        uint64 newUnlockTimestamp = uint64(block.timestamp + LOCK_DURATION);
+        uint256 lockAmount = 0;
+
+        // If the locked balance is under the target lock balance
+        if (lockedBalance < targetLockedBalanceMem) {
+            // Calculate the amount of Frax Staked Convex LP tokens to lock.
+            // The target lock balance ignoring the unlocked balance
+            uint256 targetLockAmount = targetLockedBalanceMem - lockedBalance;
```

Comment:
The only part of the strategies that is gas critical is the balance checking (since it's used in every rebase). The rest of the strategy code is used only a few times a week, and I'd rather stay out of super gas optimizations in them, and lean towards safety.